File: 3기2/디자인패턴2/최현주/SourceCode/CompositePattern/Entry.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace CompositePattern.ex2
{
    /// <summary>
    /// Component 역할
    /// Composite(복합 객체)를 위한 추상화 클래스
    /// </summary>
    public abstract class Entry
    {
        //공통적으로 파일 or 디렉토리의 이름,사이즈.
        public abstract string GetName();
        public abstract int GetSize();

        /// <summary>
        /// 복합 개체(Directory)에서만 필요한 메서드.
        /// </summary>
        /// <param name="entry"></param>
        public virtual void Add(Entry entry)
        {
        }

        //공통으로 사용되는 출력 method.
        public void Print()
        {
            PrintList("");
        }

        public abstract void PrintList(string prefix);

        public string ToString()
        {
            return string.Format("{0} ({1})", GetName(), GetSize());
        }
    }
}

File: 3기2/디자인패턴2/최혁/workspace/Visitor/src/Entry.java
import java.util.Iterator;

public abstract class Entry implements Element {
	public abstract String getName();

	public abstract int getSize();

	public Entry add(Entry entry) throws FileTreatmentException {
		throw new FileTreatmentException();
	}

	public Iterator<?> iterator() throws FileTreatmentException {
		throw new FileTreatmentException();
	}

	// public void printList() {
	// 	printList("");
	// }

	// protected abstract void printList(String prefix);

	/**
	 * getName , getSize 는 추상클래스 이지만 하위클래스에서 이 두 메소드를 구현하여 toString에서
	 * 호출하고있습니다.(Template Pattern)
	 */
	public String toString() {
		return getName() + " (" + getSize() + ")";
	}
}
